jsre/natto/natto.go

```diff
@@ -0,0 +1,130 @@
+/*
+
+This file is a modified version of https://github.com/robertkrimen/natto/blob/master/natto.go
+
+Package natto is an example/offshoot of otto that implements an event loop (supporting setTimeout/setInterval).
+
+http://godoc.org/github.com/robertkrimen/natto
+
+otto: http://github.com/robertkrimen/otto (http://godoc.org/github.com/robertkrimen/otto)
+
+*/
+package natto
+
+import (
+	"github.com/robertkrimen/otto"
+	"time"
+)
+
+type _timer struct {
+	timer    *time.Timer
+	duration time.Duration
+	interval bool
+	call     otto.FunctionCall
+}
+
+// Run will execute the given JavaScript, continuing to run until all timers have finished executing (if any).
+// The VM has the following functions available:
+//
+//      <timer> = setTimeout(<function>, <delay>, [<arguments...>])
+//      <timer> = setInterval(<function>, <delay>, [<arguments...>])
+//      clearTimeout(<timer>)
+//      clearInterval(<timer>)
+//
+func Run(vm *otto.Otto, src interface{}) (value otto.Value, err error) {
```

Comment:
I don't see a reason why this should be in a separate package. Please move this to package `jsre`.